api/swagger.yaml

```diff
@@ -173,6 +173,20 @@ tags:
     x-displayName: "System"
 
 definitions:
+  Capabilities:
+    description: "Represents engine capabilities"
+    type: "object"
+    properties:
+      v_:
+        description: "The version of capabilities this response represents."
+        type: integer
+        x-nullable: false
+      registry-client-auth:
+        description: "Whether the engine supports client auth handling."
+        type: "boolean"
+        x-nullable: true
+    example:
+      registry-client-auth: true
```

Comment:
This one needs updating to match the new output. Where possible, it's good to put example-values _per field_ that way we make sure that the example output matches the actual structure. (Swagger will blindly show whatever is in `example`, even if that doesn't match the definition; putting example values per-field makes sure that every field we define is included in the example, and that it matches the schema);


⚠️ ❓ I noticed though that the `_v` field is embedded in the `capabilities` struct, and I wonder if it should be outside of that; otherwise `_v` could "technically" be considered a capability?


Currently it produces;

```bash
curl -s --unix-socket /var/run/docker.sock http://foo/_ping?capabilities=1
{"capabilities":{"_v":1,"registry-client-auth":false}}
```


And I wonder if that should be;


```bash
curl -s --unix-socket /var/run/docker.sock http://foo/_ping?capabilities=1
{"_v":1, "capabilities":{"registry-client-auth":false}}
```

Reply:
I'd rather not do this – there's a lot of benefits to having it this way, i.e. OpenAPI support for discriminator fields, etc., as well as it makes sense for the type that holds those specific capabilities to include that version. Otherwise, the capabilities object itself does not contain enough information for a client to parse it. This only poses an issue if we want to treat capabilities as a map of arbitrary key-values, which we've already gone over and decided against.

Reply:
Ended up changing this a bit (see [here](https://github.com/moby/moby/pull/49041#discussion_r1886922941) too) – I do think the current situation might be nicer (at least, UX-wise for this kind of stuff) but it does result in:
```go
curl -s --unix-socket /var/run/docker.sock http://foo/_ping?capabilities=1
{
	"capabilities": {
		"_v": 1,
		"capabilities": { 
			"registry-client-auth": false
		}
	}
}
```